vite.config.remote.js

```diff
@@ -33,43 +33,47 @@ export default defineConfig( {
 	},
 } );
 
-function externalize( id ) {
-	const externalDefinition = defaultRequestToExternal( id );
-	return (
-		!! externalDefinition &&
-		! id.match( /\.css(?:\?inline)?$/ ) &&
-		! [ 'apiFetch', 'i18n', 'url', 'hooks' ].includes(
-			externalDefinition[ externalDefinition.length - 1 ]
-		)
-	);
+function external( id ) {
+	const hasExternal = defaultRequestToExternal( id ) !== undefined;
+	const isInlineCss = id.match( /\.css\?inline$/ );
+
+	return hasExternal && ! isInlineCss;
 }
 
 /**
  * Transform code by replacing WordPress imports with global definitions.
+ * E.g., `import { __ } from '@wordpress/i18n';` becomes `const { __ } = window.wp.i18n;`
+ * This replicates Gutenberg's behavior in a browser environment, which relies upon
+ * the `@wordpress/dependency-extraction-webpack-plugin` module.
+ *
+ * See: https://github.com/WordPress/gutenberg/tree/d2fce222ebbbef8dbc56eee1badcfe4ae0df04b0/packages/dependency-extraction-webpack-plugin
  *
  * @return {Object} The transformed code and map.
  */
 function wordPressExternals() {
 	return {
 		name: 'wordpress-externals-plugin',
-		transform( code, id ) {
+		transform( code ) {
 			const magicString = new MagicString( code );
 			let hasReplacements = false;
 
 			// Match WordPress and React JSX runtime import statements
 			const regex =
-				/import\s*(?:{([^}]+)}\s*from)?\s*['"](@wordpress\/([^'"]+)|react\/jsx-runtime)['"];/g;
+				/import\s*(?:(?:(\w+)|{([^}]+)})\s*from\s*)?['"](@wordpress\/[^'"]+|react\/jsx-runtime)['"];/g;
 			let match;
 
 			while ( ( match = regex.exec( code ) ) !== null ) {
-				const [ fullMatch, imports, module ] = match;
+				const [ fullMatch, defaultImport, namedImports, module ] =
+					match;
+				const imports = defaultImport || namedImports;
+
+				if ( module.match( /\.css\?inline$/ ) ) {
+					continue; // Exclude inlined CSS files from externalization
+				}
```

Comment:
Now that default imports matching is fixed, we must exclude inlined CSS files from externalization. There are no globals that represent inlined CSS strings.